File: src/Constants/Console.cc
#include "Constants/Console.h"
#include <string>

void print_with_color(Colors color, const char *str)
{
    printf("\033[%dm%s\033[%dm", color, str, RESET);
}
void print_with_color(Colors color, std::string str)
{
    printf("\033[%dm%s\033[%dm", color, str.c_str(), RESET);
}

std::string get_with_color(Colors color, std::string str)
{
    return "\033[" + std::to_string(color) + "m" + str + "\033[0m";
}

File: src/Controllers/Console.ts
export enum Colors {
    BLACK = 30,
    RED = 31,
    GREEN = 32,
    YELLOW = 33,
    BLUE = 34,
    MAGENTA = 35,
    CYAN = 36,
    WHITE = 37,
    RESET = 0
}


export class Console {
    static getWithColor(color: Colors, message: string) {
        return `\x1b[${color}m${message}\x1b[${Colors.RESET}m`
    }

    static printWithColor(color: Colors, message: string) {
        console.log(Console.getWithColor(color, message));
    }
}

export default {Console, Colors};